### ISSUE TITLE
Transfer Learning - Single model VS Feature extractor and single Dense

### ISSUE THREAD

**Issue by RomainCendre** [keras-team/keras#11803]
Hi everyone,
I'm here because I'm loosing my mind from two weeks...
I try to make some transfer learning on few medical images, I would like to classify them in benign or malignant, just making a training step on last new Dense layer. I proceed in a first way like this and results were not impressive:

`	# Single model
	# Clone local model
	inception_model = InceptionV3(weights='imagenet', include_top=False, pooling='avg')
	# Now we customize the output consider our application field
	prediction_layers = Dense(2, activation='softmax', name='predictions')(inception_model.output)
	# And defined model based on our input and next output
	model = Model(inputs=inception_model.input, outputs=prediction_layers)
	for layer in model.layers[:len(inception_model.layers)]:
		layer.trainable = False

	model.compile(optimizer="Adam", loss='categorical_crossentropy', metrics=['accuracy'])

	# Fit data
	model.fit(x=xtr,y=ytr, epochs=epochs, validation_data=(xte,yte), callbacks=ClassifierDeep.get_callbacks(current_dir))
`

But when I try to do it in two steps, everything seems to going fine:

```
################
	# Two way model
	# Clone local model
	features = InceptionV3(weights='imagenet', include_top=False, pooling='avg')
	for layer in features.layers:
		layer.trainable = False
	classif = Sequential()
	classif.add(Dense(2, activation='softmax', name='predictions'))
	classif.compile(optimizer="Adam", loss='categorical_crossentropy', metrics=['accuracy'])

	# Fit data
	classif.fit(x=features.predict(xtr), y=ytr, epochs=epochs, validation_data=(features.predict(xte),yte), callbacks=ClassifierDeep.get_callbacks(current_dir))
```
Can someone give me a clue?
Best regards

**Comment by RomainCendre** [comment]
Perhaps is this due to Batch Normalization as referred by #9965 and #9214?
